### DISCUSSION TITLE
RasterioSource Loading Warning

### DISCUSSION THREAD

**Discussion by ammarsani-my** [azavea/raster-vision#1598] [Q&A]
Hi @AdeelH @lewfish

I'm trying to load my orthophoto like this:

```
from rastervision.core.data import RasterioSource
orthomosaic = RasterioSource("some_path/orthomosaic.tif")
```

Then this warning appeared:
```
rastervision.core.data.raster_source.rasterio_source: WARNING - Raster block size (32, 15500) is too non-square. This can slow down reading. Consider re-tiling using GDAL.
```

But if I query this `orthomosaic.shape` it returns `(16460, 15500, 3)`. Seem like not having issue.

But then I tried to reverse engineering the code then found out the root of this number came by running this: `set(orthomosaic.image_dataset.block_shapes)`. It return `{(32, 15500)}`.

May I know why this warning showed up? Do I need to re-tiling? If need, may I know how to do this?

Thanks!

**Comment by AdeelH** [comment]
The warning is about the block size that [the GeoTIFF format uses to store the raster](https://gdal.org/drivers/raster/gtiff.html#creation-options), not the size of the full raster.

See https://gis.stackexchange.com/questions/401818/gdal-change-band-block-size.
